### PR TITLE
CXX-628: create unreachable macro for bsoncxx

### PR DESCRIPTION
**Pull request by Machyne**

### CODE REVIEW

File: src/bsoncxx/config/prelude.hpp [hunk 1]
```diff
@@ -16,3 +16,7 @@
 #include <bsoncxx/export.hpp>
 
 #define BSONCXX_INLINE inline __attribute__ ((__visibility__("hidden"), __always_inline__))
+
+#pragma push_macro("BSONCXX_UNREACHABLE")
+#undef BSONCXX_UNREACHABLE
+#define BSONCXX_UNREACHABLE std::abort();
```
Comment (amidvidy) [inline src/bsoncxx/config/prelude.hpp]: can you take off the semicolon here, and then put it after uses of BSONCXX_UNREACHABLE? I think it looks cleaner that way and we generally prefer to make our macros statement-like

Comment (Machyne) [inline src/bsoncxx/config/prelude.hpp]: Fixed!

**Comment by amidvidy** [comment]
LGTM

**Comment by Machyne** [comment]
@samantharitter - good to merge?

**Comment by samantharitter** [comment]
LGTM

**Comment by acmorrow** [comment]
LGTM

**Pull request merged**